docs/StateTransitionLogger.md

```diff
@@ -0,0 +1,83 @@
+
+We will add another first-class process type to the Kōjō complement (e.g. `Server`, `Root`) called the `StateTransitionLogger` (working title).
+The `StateTransitionLogger` will be a child of the `Root`, and will be most like `Worker` processes.
+There will only be one `StateTransitionLogger` per Kōjō cluster, in the same way that there is only one `Worker` acting as the `Maintainer`.
+`Root`s will be responsible for babysitting the status of the `StateTransitionLogger` (by inserting `command.addProcess('state_transition_logger')` messages into the "publication" redis list if nothing is holding the mutex).
+Once the `StateTransitionLogger` is instantiated, it will poll the `kojo_job_state_transitions` table continuously.
+For each transition event the `StateTransitionLogger` pulls into memory, it will emit a message and then delete that row.
+This guarantees at-least-once delivery of transition messages.
+
+## Backward Incompatible Changes
+If there are any assumptions within Kōjō about the process hierarchy, they could be violated by the addition of a new process type, but this is unlikely.
+Otherwise it's a purely additive modification to Kōjō.
+
+## Example 1
+1. There exists a dynamically scheduled job with `previous_state: 'new'`, `assigned_state: 'waiting'`, `next_state_request: 'working'`, and `work_at_datetime < NOW()`
+1. A recently spawned `Worker` process selects that job to work
+1. That `Worker` process reaches `Neighborhoods\Kojo\Foreman::_updateJobAsWorking()` and invokes `Neighborhoods\Kojo\State\Service::applyRequest()` to transition the job from `waiting` to `working`
+1. The job, process, and transition information are inserted into `kojo_job_state_transitions`
+1. The `Worker` process continues execution and hands over control to userspace
+1. Concurrently, the `StateTransitionLogger` process for that cluster (not necessarily in the same execution environment) queries `kojo_job_state_transitions` and pulls that transition information into memory.
+1. The `StateTransitionLogger` emits a message, deletes the row, and moves on to the next transition event
+1. Concurrently, our logging infrastructure consumes the emitted messages
+
+## Future Scope
+Part of the design for the `StateTransitionLogger` is to delegate more one-per-cluster responsibilities to first-class processes.
+This is in contrast to the typical Kōjō pattern of requiring every newly spawned `Worker` process to attempt to perform these responsibilities.
+If this implementation of the `StateTransitionLogger` is successful, it would be desireable to refactor out `Maintainer`, `Scheduler`, etc. responsibilities in the same way.
+
+## Drawbacks
+There are multiple unknowns with this design which could cause it to become infeasible to implement, in which case we'd have to go with the approach outlined in Alternative 3.
+
+## Unresolved Questions
+* Should we log when a job is created (i.e. scheduled)?
```

Comment:
On the surface I would answer "yes". When an execution cluster is constrained on resources or limited by process pool size, there may be jobs that never exit the `waiting` state. In this condition, we'd never see a `StateTransitionLogger` message about that job, and only polling `kojo_job` would display it.

Is there an implementation detail that makes logging on creation difficult?

Reply:
After we talked through the implementation a little bit last week, I realized that including job creation is easier than not including it